cpp/Dockerfile.ubuntu-bionic

```diff
@@ -17,78 +17,102 @@
 
 FROM ubuntu:bionic
 
+# pipefail is enabled for proper error detection in the `wget | apt-key add`
+# step
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
-ENV DEBIAN_FRONTEND noninteractive
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update -y -q && \
+    apt-get install -y -q --no-install-recommends \
+      wget software-properties-common gpg-agent && \
+      apt-get clean && rm -rf /var/lib/apt/lists*
+
+# Installs LLVM toolchain, for gandiva and testing other compilers
+#
+# Note that this is installed before the base packages to improve iteration
+# while debugging package list with docker build due to slow download speed of
+# llvm compared to ubuntu apt mirrors.
+ARG LLVM_VERSION=7
+# Args are only exposed in the "build" step, this ensure that LLVM_VERSION is
+# found in the "run" step.
+ENV LLVM_VERSION=${LLVM_VERSION}
+ARG LLVM_APT_URL="http://apt.llvm.org/bionic/"
+ARG LLVM_APT_ARCH="llvm-toolchain-bionic-${LLVM_VERSION}"
+RUN wget -q -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
+RUN apt-add-repository -y --update "deb ${LLVM_APT_URL} ${LLVM_APT_ARCH} main"
+RUN apt-get install -y -q --no-install-recommends \
```

Comment:
Shrink these three steps into a single one.

Reply:
The goal is to make the build phase much faster when one doesn't change LLVM_VERSION, the packages are huge and slow to download. I think we should keep it as-is.

Reply:
Note, I did this because I toyed often with removing/adding packages in the list while refatoring.

Reply:
I agree with combining the three RUN steps into one

Reply:
Will do.